src/moonlink_service/src/test.rs

```diff
@@ -1108,3 +1135,67 @@ async fn test_create_table_with_invalid_config() {
         .unwrap();
     assert!(!response.status().is_success());
 }
+
+#[tokio::test]
+#[serial]
+async fn test_schema_invalid_struct_missing_fields() {
+    let _guard = TestGuard::new(&get_moonlink_backend_dir()).await;
+    let config = get_service_config();
+    tokio::spawn(async move {
+        start_with_config(config).await.unwrap();
+    });
+    test_readiness_probe().await;
+
+    let client = reqwest::Client::new();
+    let crafted_src_table_name = format!("{DATABASE}.invalid-struct-table");
+    let payload = json!({
+        "database": DATABASE,
+        "table": "invalid-struct-table",
+        "schema": [
+            {"name": "id", "data_type": "int32", "nullable": false},
+            {"name": "bad_struct", "data_type": "struct", "nullable": true}
+        ],
+        "table_config": {"mooncake": {"append_only": true}}
+    });
+
+    let response = client
+        .post(format!("{REST_ADDR}/tables/{crafted_src_table_name}"))
+        .header("content-type", "application/json")
+        .json(&payload)
+        .send()
+        .await
+        .unwrap();
+    assert!(!response.status().is_success());
```

Comment:
suggest to include the `status` in the assertion, so at least we know what happened on failure